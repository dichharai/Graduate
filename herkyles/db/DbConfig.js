import * as firebase from "firebase";

const firebaseConfig = {
    apiKey: "AIzaSyD5Ha8fSPEBkjqMchNVy8zU6vB-WvcGYTM",
    authDomain: "herkyles-85e8e.firebaseapp.com",
    databaseURL: "https://herkyles-85e8e.firebaseio.com",
    projectId: "herkyles-85e8e",
    storageBucket: "herkyles-85e8e.appspot.com",
    messagingSenderId: "812934741247"
};

export const firebaseApp = firebase.initializeApp(firebaseConfig); 