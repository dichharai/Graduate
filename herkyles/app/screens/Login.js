import React, { Component } from 'react'; 
import { View, Text, StyleSheet, AsyncStorage, Button, KeyboardAvoidingView} from 'react-native'; 
import {firebaseApp} from '../../db/DbConfig';

import {Container} from '../components/Container'; 
import {TitledInput} from '../components/TitledInput'; 
import {Spinner} from '../components/Spinner';  
import ErrorStatus from '../components/Status/ErrorStatus'; 
import SuccessStatus from '../components/Status/SuccessStatus'; 
import {LinkBtns} from '../components/Buttons/LinkBtns';
import {LinkTouch} from '../components/Buttons/LinkTouch'; 


class Login extends Component{
    static navigationOptions = {
        title: "Login", 
        headerStyle: {
            backgroundColor: "#000000",
        }, 
          headerTintColor: '#facf33',
    };
    
    constructor(props){
        super(props); 
        this.state = {email: '', password: '', error: '', loading: false, signupToLoginStatus: ''}; 
        this._getStatus(); 
    }; 
    _getStatus = async () =>{
        try{
            let status = await AsyncStorage.getItem('signupToLogin');
            console.log("status in login: " + status);
            this.setState({signupToLoginStatus: status});
            console.log("status from state: "+ this.state.signupToLoginStatus);
            
        }catch(e){
            console.log(e);
        }
    };
    
    _signInAsync = async (email) => {
        const userTokenValue = Math.round((Math.random()*1000));

        await AsyncStorage.setItem("userToken", userTokenValue.toString());
        await AsyncStorage.setItem("userEmail", email);

        let userToken = await AsyncStorage.getItem("userToken");
        console.log("userToken after setting: " + userToken);
        
        this.props.navigation.navigate('Auth');
    }; 
    _removeSignupToLoginStatus = async () => {
        await AsyncStorage.removeItem('signupToLogin'); 
    }; 


    onLoginPress = () => {
        this.setState({ error: '', loading: true, signupToLoginStatus: '' });
        {this._removeSignupToLoginStatus()}; 
        const { email, password } = this.state;
        firebaseApp.auth().signInWithEmailAndPassword(email, password)
            .then(() => { this.setState({ error: "", loading: false });
                console.log("Logged in successfully");
                this._signInAsync(email);
                       
            })
            .catch((error) => {
                //Login was not successful, give an error message
                console.log("In error catch scope");
                console.log(error.code);
                this.setState({ error: 'Authentication failed.', loading: false });
                
            });
    }; 
    renderButtonOrSpinner(){

        if (this.state.loading) {
            return <Spinner />    
        }; 
        
          
        return <LinkBtns  
                text="Log in"
                onPress={this.onLoginPress}  />;
    }
    renderSuccessStatus = () => {
        if(this.state.signupToLoginStatus){
            return <SuccessStatus text={this.state.signupToLoginStatus}/>
        }; 
    }
   
   handleSignupPress = () => {
        this.props.navigation.navigate('Signup'); 
        console.log("Sign up pressed"); 
    }

    convertToUpperCase = (label) => {
        return label.toUpperCase(); 
    }

    render(){
        return (
            <Container>
                <TitledInput
                    label= {this.convertToUpperCase('Email Address')}
                    placeholder="you@domain.com"
                    value={this.state.email}
                    onChangeText={email =>this.setState({email})}
                    secureTextEntry={false}
                />
                 <TitledInput
                    label= {this.convertToUpperCase("Password")}
                    placeholder="********"
                    value={this.state.password}
                    onChangeText={password =>this.setState({password})}
                    secureTextEntry={true}
                />
                <ErrorStatus 
                    text={this.state.error} 
                />

                {this.renderSuccessStatus()}
                {this.renderButtonOrSpinner()}

                <LinkTouch
                    text="No account? Signup" onPress={this.handleSignupPress}
                />
 
            </Container>            
        );
    }
}
export default Login; 