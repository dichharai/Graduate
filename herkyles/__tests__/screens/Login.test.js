import React from "react";
import Login from "../../app/screens/Login";

import renderer from "react-test-renderer";
import * as firebase from '../../db/DbConfig';

describe ("Login", () => {
  it("renders without crashing", () => {
    const rendered = renderer.create(<Login />).toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });
  it('should call navigate on pressing Signup', () => {
    let handleLoginPressMock = jest.fn(); 
    //let navigationMock = {navigate: jest.fn()}
    const navigationMock = {navigate: jest.fn()}; 

    let loginComponent = renderer.create(<Login 
        navigation={navigationMock}
      />
    ).getInstance();
  
    //loginComponent.handleSignupPress(); 
    //expect(navigationMock.navigate).toHaveBeenCalledTimes(1); 
  });
  
});