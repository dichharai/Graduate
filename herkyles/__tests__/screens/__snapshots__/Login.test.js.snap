// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={undefined}
  testID={undefined}
>
  <View
    style={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}
    >
      EMAIL ADDRESS
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="you@domain.com"
      secureTextEntry={false}
      style={undefined}
      value=""
    />
  </View>
  <View
    style={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}
    >
      PASSWORD
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="********"
      secureTextEntry={true}
      style={undefined}
      value=""
    />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        Log in
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        No account? Signup
      </Text>
    </View>
  </View>
</View>
`;
